apps/ca.c

```diff
@@ -795,7 +795,7 @@ int ca_main(int argc, char **argv)
      */
     if (def_ret == 2 && def_nid == NID_undef) {
         /* The signing algorithm requires there to be no digest */
-        dgst = EVP_md_null();
+        dgst = (EVP_MD *)EVP_md_null();
```

Comment:
This proliferation of casts has me wonder if we should un-constify the likes of `EVP_md_null()`, `EVP_get_digestbyname()`, ...

Reply:
(the cast could be done inside those functions)

Reply:
I would not make `EVP_md_null` etc non-const. It is a transition for moving to a pure-fetch environment.

Reply:
Yeahok...  I can't help but find it jarring, but you do have a point.

Reply:
Unconstifying would break existing callers so that is another reasons not to do that.